src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String ARRAY_LIST_IOOB_EXCEPTION_TEXT
+            = "No element with such index";
+    private static final int WRONG_INDEX = -1;
+    private T[] data;
+    private int size;
+
+    public ArrayList() {
+        this.data = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == data.length) {
+            grow();
+        }
+        data[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(ARRAY_LIST_IOOB_EXCEPTION_TEXT);
```

Comment:
You have a mathod "checkIndex" for this

Reply:
yep, but this statement is a bit different

Reply:
Yeah, but we can write if(index != size) and then use our method, I think it will be more readable